custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/OnlyTabIdentationInXmlFilesCheck.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.XML_EXTENSION;
+
+import java.io.File;
+
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * Checks whether whitespace characters are use instead of tabs in xml files 
+ * indentations and generates warnings in such cases.
+ * 
+ * @author Lyubomir Papazov - initial contribution
+ *
+ */
+public class OnlyTabIdentationInXmlFilesCheck extends AbstractStaticCheck {
+
+    private static final String PATTERN_TO_BE_FOLLOWED = "^\\t*[<].*";
```

Comment:
This regex is just wrong, should you maybe rather check on lines starting with spaces?

According to this regex you can not have lines with multi-line CDATA blocks, you can not split your header:

```
<thing:thing-descriptions bindingId="onewiregpio" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 
        http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
```

Reply:
And quickly reading its not even allowed to have empty lines?

Reply:
I didn't think of those cases. I'll have a look at it asap.